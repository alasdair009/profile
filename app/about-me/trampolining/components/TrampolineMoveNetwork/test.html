<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
