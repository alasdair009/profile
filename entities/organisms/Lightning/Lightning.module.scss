@use "../../../app/styles/animation";
@use "../../../app/styles/colors";
@use "../../../app/styles/dimensions";
@use "../../../app/styles/helpers/rem";

@keyframes strikeAnimation {
  0%,
  49%,
  55%,
  100% {
    opacity: 0;
  }
  50% {
    opacity: 1;
  }
}

@keyframes strikeGlow {
  0%,
  53%,
  73%,
  100% {
    filter: brightness(1);
  }
  54% {
    filter: brightness(3);
  }
}

.root {
  position: relative;
}

.inner {
  display: flex;
  height: 100vh;
  margin: 0 auto;
  max-width: rem.toRem(dimensions.$globalDecorationMaxWidth);
  overflow: hidden;
  position: relative;
  width: 100%;
}

.background {
  animation: strikeGlow var(--frequency) linear infinite;
  height: 100%;
  left: 0;
  object-fit: cover;
  position: absolute;
  top: 0;
  width: 100%;

  @media (prefers-reduced-motion) {
    animation: none;
  }
}

.strike {
  animation: strikeAnimation var(--frequency) linear infinite;
  height: 100%;
  left: 0;
  object-fit: cover;
  opacity: 0;
  position: absolute;
  top: 0;
  width: 100%;

  @media (prefers-reduced-motion) {
    animation: none;
    opacity: 1;
  }
}

.content {
  align-items: center;
  background: linear-gradient(
    to left,
    transparent,
    rgba(colors.$colorblackEvil, 0.2),
    transparent
  );
  display: flex;
  justify-content: center;
  margin: 0 auto;
  max-width: rem.toRem(globalContentMaxWidth);
  overflow: hidden;
  padding: dimensions.getSize(s8);
  width: 100%;
  z-index: 1;
}
