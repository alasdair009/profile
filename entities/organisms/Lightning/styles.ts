"use client";
import styled, {keyframes} from "styled-components";
import {globalDecorationMaxWidth} from "@/entities";
import {rem} from "polished";
import Image from "next/image";
import rainImage from "./assets/rain.png";

const rain = keyframes`
    from {
        background-position: 0 0;
    }

    to {
        background-position: -20% -100%;
    }
`;
export const Root = styled.section`
    position: relative;
`;

export const Inner = styled.div`
    display: flex;
    height: 100vh;
    margin: 0 auto;
    max-width: ${rem(globalDecorationMaxWidth)};
    overflow: hidden;
    position: relative;
    width: 100%;

    &::before {
        animation: ${rain} 0.3s linear infinite;
        background-image: url(${rainImage.src});
        content: "";
        height: 100%;
        left: 0;
        position: absolute;
        top: 0;
        transform: rotate(180deg);
        width: 100%;
    }
`;

export const Background = styled(Image)`
    height: 100%;
    object-fit: cover;
    width: 100%;
`;

export const Strike = styled(Image)`
    height: 100%;
    left: 0;
    object-fit: cover;
    position: absolute;
    top: 0;
    width: 100%;
`;