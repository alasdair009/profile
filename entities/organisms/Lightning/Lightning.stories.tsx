import { Lightning } from "./Lightning";
import { Meta, StoryObj } from "@storybook/react";
import { expect, within } from "@storybook/test";

const meta: Meta<typeof Lightning> = {
  component: Lightning,
  parameters: {
    controls: { hideNoControlsWarning: true },
  },
  args: {
    frequency: 2,
  },
  argTypes: {
    frequency: {
      control: { type: "range", min: 1, max: 3, step: 0.1 },
    },
  },
};
export default meta;

export const Default: StoryObj<typeof Lightning> = {
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const lightningElement = canvas.getByTestId(Lightning.name);

    await expect(lightningElement).toBeInTheDocument();
  },
};
