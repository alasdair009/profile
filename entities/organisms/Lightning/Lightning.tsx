import { CSSProperties, HTMLAttributes } from "react";
import backgroundImage from "./assets/background.webp";
import strikeImage from "./assets/lightining.png";
import { Rain } from "../../atoms/Rain";
import styles from "./Lightning.module.scss";
import Image from "next/image";

type LightningProps = {
  frequency?: number;
} & HTMLAttributes<HTMLDivElement>;
export function Lightning({
  frequency = 2,
  children,
  ...rest
}: LightningProps) {
  return (
    <section className={styles.root} data-testid={Lightning.name} {...rest}>
      <div className={styles.inner}>
        <Image
          className={styles.background}
          src={backgroundImage}
          alt={"Park background"}
          style={
            {
              "--frequency": `${frequency}s`,
            } as CSSProperties
          }
        />
        <Rain />
        <Image
          className={styles.strike}
          src={strikeImage}
          alt={"Rain"}
          fill={true}
          style={
            {
              "--frequency": `${frequency}s`,
            } as CSSProperties
          }
        />
        <div className={styles.content}>{children}</div>
      </div>
    </section>
  );
}

Lightning.Rain = Rain;
