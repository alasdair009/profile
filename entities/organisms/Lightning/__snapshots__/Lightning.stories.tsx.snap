// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organisms/Lightning Default smoke-test 1`] = `
<section class="sc-dKsqdn hYkMkV">
  <div class="sc-eowDPD cBJDOg">
    <img alt="Park background"
         loading="lazy"
         width="1348"
         height="654"
         decoding="async"
         data-nimg="1"
         class="sc-ezTrPE gApgwz"
         srcset="/static/media/entities/organisms/Lightning/assets/background.webp?w=1920&amp;q=75 1x, /static/media/entities/organisms/Lightning/assets/background.webp?w=3840&amp;q=75 2x"
         src="/static/media/entities/organisms/Lightning/assets/background.webp?w=3840&amp;q=75"
         style="color: transparent;"
    >
    <div class="sc-iAlELC gRILcD">
      <canvas class="sc-oeqTF hjFjiZ">
      </canvas>
    </div>
    <img alt="Rain"
         loading="lazy"
         decoding="async"
         data-nimg="fill"
         class="sc-drnuxz jnPhzL"
         sizes="100vw"
         srcset="/static/media/entities/organisms/Lightning/assets/lightining.png?w=640&amp;q=75 640w, /static/media/entities/organisms/Lightning/assets/lightining.png?w=750&amp;q=75 750w, /static/media/entities/organisms/Lightning/assets/lightining.png?w=828&amp;q=75 828w, /static/media/entities/organisms/Lightning/assets/lightining.png?w=1080&amp;q=75 1080w, /static/media/entities/organisms/Lightning/assets/lightining.png?w=1200&amp;q=75 1200w, /static/media/entities/organisms/Lightning/assets/lightining.png?w=1920&amp;q=75 1920w, /static/media/entities/organisms/Lightning/assets/lightining.png?w=2048&amp;q=75 2048w, /static/media/entities/organisms/Lightning/assets/lightining.png?w=3840&amp;q=75 3840w"
         src="/static/media/entities/organisms/Lightning/assets/lightining.png?w=3840&amp;q=75"
         style="position: absolute; height: 100%; width: 100%; inset: 0px; color: transparent;"
    >
    <div class="sc-hDcvty elwTpw">
    </div>
  </div>
</section>
`;
